### PR TITLE
[KernelGen] Add sparse_attn_indexer for DeepSeek V4 MLA prefill

### PR DESCRIPTION
**Pull request by yzw1128**
# [KernelGen] Add sparse_attn_indexer fused operator for DeepSeek V4 MLA

## Summary
Add `sparse_attn_indexer` Triton operator for DeepSeek V4 sparse attention prefill phase. This kernel selects the top-K KV cache positions for each query token, replacing the vLLM tilelang-based CUDA implementation.

## Motivation
- Critical operator in DeepSeek V4 MLA sparse attention pipeline
- Selects which KV cache positions each query token should attend to
- Triton implementation achieves 3x+ speedup over vLLM CUDA for decode and small prefill

## Implementation
**Generated via KernelGen**

This implementation was developed through 12 iterative optimization rounds:
- Started from vLLM's tilelang CUDA kernel as baseline
- Key optimizations discovered:
  - **Compact logits tensor**: Allocate (active_tokens × max_kv_end) instead of (active_tokens × total_kv_len), reducing memory up to 4x for typical configs
  - **3-path algorithm**: Shortcut fill for trivial tokens (kv_end ≤ topk), fused single-kernel for small KV, multi-kernel for large KV
  - **Binary search threshold top-K**: O(N·log(range)) replaces O(N·K) insertion sort
  - **Parallel collect with atomic counters**: Eliminates sequential bottleneck in index gathering
  - **FP8 E4M3 quantized KV cache**: Per-token scale stored as 4 LE bytes alongside key data

### FP8 Quantization Format
Each KV cache slot (head_dim + 4 bytes):
- `[0:head_dim]`: FP8 E4M3 quantized key vector
- `[head_dim:head_dim+4]`: FP32 scale as little-endian uint8 bytes

### Scoring Formula
```
logit[token_i, kv_pos_j] = sum_h(ReLU(q_h · k_j) * weight_h) * k_scale_j
```

### Top-K Selection (binary search + collect)
1. Find min/max of logits
2. Binary search for threshold where count(logits ≥ threshold) = topk
3. Parallel collect all positions exceeding threshold

## Testing
- ✅ Pytest test suite with pure-PyTorch reference (no vLLM dependency)
- ✅ Tests cover shortcut path, top-K selection, K cache insertion
- ✅ Set-based comparison (order-independent) with 95% match threshold
- ✅ Configurations: num_tokens ∈ {1, 32, 128}, max_model_len ∈ {2048, 4096}

## Performance
DeepSeek V4 config: num_heads=64, head_dim=128, topk=1024

| Configuration | Triton (μs) | CUDA (μs) | Speedup |
|---------------|-------------|-----------|---------|
| 1 token / 4096 ctx | 23.3 | 72.9 | 3.13x |
| 128 tokens / 4096 ctx | 21.6 | 67.1 | 3.11x |
| 2048 tokens / 8192 ctx | 922.8 | 198.3 | 0.21x |

The kernel excels at decode (single token) and small prefill scenarios which dominate inference workloads. Large prefill (2048 tokens) is bounded by logit computation over the full KV range.

## Files Changed
- `src/flag_gems/fused/DSA/sparse_attn_indexer.py`: Main Triton kernel implementation
- `src/flag_gems/fused/__init__.py`: Export sparse_attn_indexer
- `conf/operators.yaml`: Register operator
- `tests/test_DSA/test_sparse_attn_indexer.py`: Pytest correctness tests
- `benchmark/test_sparse_attn_indexer.py`: Performance benchmark

### CODE REVIEW

File: src/flag_gems/fused/sparse_attn_indexer.py [hunk 1]
```diff
@@ -0,0 +1,584 @@
+"""Sparse attention indexer for DeepSeek V4 MLA prefill.
```
Comment (0x45f) [inline src/flag_gems/fused/sparse_attn_indexer.py:1]: move file to fused dir

File: tests/test_sparse_attn_indexer.py [hunk 2]
```diff
@@ -0,0 +1,473 @@
+"""Correctness tests for sparse_attn_indexer Triton kernel.
```
Comment (0x45f) [inline tests/test_sparse_attn_indexer.py:1]: move to test dir

Comment (0x45f) [inline tests/test_sparse_attn_indexer.py:1]: can we use `utils.gems_assert_equal` to check results?

**Comment by bin913** [comment]
@yzw1128 Please fix failing checks

**Comment by huangyiqun** [comment]
1. Plz fix code style.
2. Plz fix python-op CI test.https://github.com/flagos-ai/FlagGems/actions/runs/25909958425/job/76514944224?pr=3280

**Comment by bin913** [comment]
@yzw1128 Please resolve conflicts

**Comment by w1120029931-bit** [comment]
Is the PR still being followed up？
